src/automata/FireAndForgetRequester.h

```diff
@@ -0,0 +1,55 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+#pragma once
+
+#include <memory>
+
+#include "src/Payload.h"
+
+#include "src/ReactiveStreamsCompat.h"
+#include "src/mixins/ConsumerMixin.h"
+#include "src/mixins/ExecutorMixin.h"
+#include "src/mixins/LoggingMixin.h"
+#include "src/mixins/MemoryMixin.h"
+#include "src/mixins/MixinTerminator.h"
+#include "src/mixins/PublisherMixin.h"
+#include "src/mixins/SinkIfMixin.h"
+#include "src/mixins/SourceIfMixin.h"
+#include "src/mixins/StreamIfMixin.h"
+
+namespace reactivesocket {
+
+/// Implementation of an automaton that represents a Fire-and-Forget requester.
+/// Note that it's simpler than other automata due to the inherent throwaway
+/// simplicity of Fire-and-Forget semantics. Stream IDs are not tracked.
+///
+/// TODO: we need to add logging similar to other *Requester automata.
+class FireAndForgetRequester: public Subscription {
```

Comment:
For discussion:

I opted out of mixins for FireAndForgetRequester because of its simple semantics. The logic follows the logic in the Java implementation. Are we missing certain properties that would require the mixins? Do we need ExecutorMixin?